src/server/fileio.js

```diff
@@ -35,9 +37,59 @@ export const configurableUploadHandler = (req, res, apiBaseURL) => {
         })
     ).pipe(res);
 };
+
 /**
  * Same as configurableUploadHandler, but defaults to using the configured
  * terrain handler.
  */
 export const uploadHandler = (req, res) =>
     configurableUploadHandler(req, res, terrainURL);
+
+/**
+ * Handles proxying file download requests to the Terrain API.
+ * @param {object} req - An Express Request object.
+ * @param {object} res - An Express Response object.
+ * @param {string} apiBaseURL - The base URL to the backing API. Usually an instance of the terrain API.
+ * @returns null
+ */
+export const configurableDownloadHandler = (req, res, apiBaseURL) => {
+    const token = req?.user?.accessToken;
+
+    if (!token) {
+        res.status(401).send("Authorization required.");
+        return;
+    }
+
+    const { source } = req.query;
+    if (!source || source === "") {
+        res.status(500).send("source query parameter must be set.");
```

Comment:
A super tiny thing that doesn't really matter much, but this should probably be a 400 (client error) rather than 500 (server error)

Reply:
Good catch, thanks. I'll push a fix soon.